### PR TITLE
fix: catalogue queries

### LINKED ISSUE #2229: UMCG Research Data Catalogue 'areas of information' filter broken
**What version of EMX2 are you using (see footer)**
[8.172.0](https://github.com/molgenis/molgenis-emx2/releases/tag/v8.172.0)

**Describe the bug**
Areas of information filter doesn't show any filter options

**To Reproduce**
1. Go to 'https://umcgresearchdatacatalogue.nl/UMCG/ssr-catalogue/cohorts/'
2. Open Areas of information filter
3. No options are shown

Other filters still work; areas of information ontology is filled; forms work as well as expected.

**Screenshots**
filter

### PR DESCRIPTION
**Pull request by mswertz**
* feat: search terms should be applied using 'or' on subfilters, in this case 'root', 'collectionEvents' and 'subcohorts'
* fix: in graphql, when settings 'search' in the root of a filter tree it was not applied (needed for the 'or' mentioned above)

### CODE REVIEW

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/cohorts/index.vue [hunk 1]
```diff
@@ -19,12 +19,11 @@ let filters = reactive([
     title: "Search in cohorts",
     columnType: "_SEARCH",
     search: "",
-    searchTables: ["collectionEvents", "subcohorts"],
```
Comment (connoratrug) [inline apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/cohorts/index.vue:22]: why are these removed and hardcoded in the function ? the idea is to have a config data struct that controles the filters ( and could be moved into config or db

File: backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java [hunk 2]
```diff
@@ -385,6 +385,8 @@ private List<Condition> jsonFilterQueryConditions(
         } else if (Operator.AND.equals(f.getOperator())) {
           conditions.add(
               and(jsonFilterQueryConditions(table, column, tableAlias, subAlias, f, searchTerms)));
+        } else if (TRIGRAM_SEARCH.equals(f.getOperator())) {
+          conditions.add(jsonSearchConditions(table, TypeUtils.toStringArray(f.getValues())));
```
Comment (connoratrug) [inline backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java:389]: how does this change the search behaviour ? , i'm missing the before/after test change that showt the behavioural change

Comment (mswertz) [reply backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlQuery.java:389]: The only new test included should fail before change. What this does is add filter clause for search that previously was ignored.

**Comment by connoratrug** [comment]
Looks good , think we need more test / examples in the docs that explain the different search options. if i compare with my pref PR the generated query seems the same idea ( except for the wrapping _and , i assume the AND is the default clause for combining queries, is it need ? )
Screenshot 2023-04-12 at 13 03 55

**Review by connoratrug — APPROVED** [review]

**Pull request merged**